ee/agent/knapsack/knapsack.go

```diff
@@ -184,6 +184,10 @@ func (k *knapsack) Dt4aInfoStore() types.KVStore {
 	return k.getKVStore(storage.Dt4aInfoStore)
 }
 
+func (k *knapsack) WindowsUpdatesCacheStore() types.KVStore {
```

Comment:
Hrm. Given that this is only exposed inside `tables/windowsupdatetable` it feels weird to plumb it through knapsack -- it's not crossing any package boundaries. 

It _feels_ like a package level variable would be simpler. But I know sometimes that's messy to test around. (Since it's sorta global state)

Unclear to me if this is cleaner cached on the table or in the underlying windoewsupdate pkg

Reply:
I think if we really do want to push through knapsack, I think we should make a general `cache` thing, and not make a specific one per cache. But maybe I'm premature

Reply:
I wanted the cache to persist between launcher restarts, which is why I didn't put it in the table or the windowsupdate package. I agree it's weird to have it available everywhere when it's Windows-specific...I will think on a general `cache` (maybe something similar to `InitialResultsStore`?) or maybe moving it to kv.sqlite or something!

Reply:
Sorry, I was being confusing...

I think there are 2 questions here. One is what the underlying storage is (and I assumed you went with bbolt to persist across reboots)

The other is how we connect all the dots. One way us to expose the bbolt in knapsack, that feels like it's adding a lot of boilerplate. Another way would be to pass a bbolt store to the actor, and have that stash a package level variable.

Reply:
I think if we do the latter (pass a bbolt store to the actor) then we have a direct bbolt usage, which means we lose the abstraction that we've built on top of bbolt in order to eventually replace it? Or am I misunderstanding the suggestion? (We can chat about it in our meeting tomorrow if that's easier!)

Reply:
Misunderstanding -- could pass a store to the actor, the actor can set a package level var, and we can skip knapsack.

Reply:
We currently are passing the store to the actor: `func NewWindowsUpdatesCacher(cacheStore types.GetterSetter, cacheInterval time.Duration, slogger *slog.Logger) *windowsUpdatesCacher`